src/System.Data.Common/src/System/Data/Common/DbCommand.cs

```diff
@@ -175,9 +178,12 @@ protected virtual Task<DbDataReader> ExecuteDbDataReaderAsync(CommandBehavior be
                 }
                 catch (Exception e)
                 {
-                    registration.Dispose();
                     return Task.FromException<DbDataReader>(e);
                 }
+                finally
+                {
+                    registration.Dispose();
```

Comment:
Wouldn't this cause the dispose even when the task is executing and take away the ability of the task to be cancelled?